src/coords.cpp

```diff
@@ -0,0 +1,64 @@
+#include <s2/s2latlng.h>
+#include <s2geography.h>
+
+#include <iostream>
+#include <stdexcept>
+
+#include "geography.hpp"
+#include "pybind11.hpp"
+
+namespace py = pybind11;
+namespace s2geog = s2geography;
+using namespace spherely;
+
+double get_y(PyObjectGeography a) {
+    auto geog = a.as_geog_ptr();
+    if (geog->geog_type() != GeographyType::Point) {
+        throw py::value_error("Only Point geometries supported");
+    }
+    auto point = static_cast<Point*>(geog);
+    auto s2point = point->s2point();
+    auto latlng = S2LatLng(std::move(s2point));
+    double lat = latlng.lat().degrees();
+    return lat;
```

Comment:
We could also just call `s2geography::s2_y`.

Reply:
Yes, at the time of writing this PR I didn't realize that existed, in the meantime I had noticed that as well. 
I think that also has the behaviour of returning NaN for non-point geometries, as you suggested, if we would use it as is, without any validation before calling it.

I am not entirely following what the implementation is doing though: https://github.com/paleolimbot/s2geography/blob/5b06341403ec2e030b2a254bfa96d8c3386296f3/src/s2geography/accessors.cc#L165-L178 
When is it possible that you need that for loop in there? Or does that mean that if you have a collection of two points, it will just give you the x/y of the first point? (I would find that a bit unexpected behaviour)

Reply:
Yes I was a bit confused by the implementation too. Probably safer to keep raising an error in case of non-point geographies for now.